__init__.py

```diff
@@ -288,23 +327,19 @@ def __init__(self):
         self.CODES['50d', '50n'] = 7                # windy/misty
 
         # Use Mycroft proxy if no private key provided
-        self.settings["api_key"] = None
-        self.settings["use_proxy"] = True
+        self.settings.setdefault("api_key", None)
+        self.settings.setdefault("use_proxy", True)
 
     def initialize(self):
-        # TODO: Remove lat,lon parameters from the OWMApi()
-        #       methods and implement _at_coords() versions
-        #       instead to make the interfaces compatible
-        #       again.
-        #
-        # if self.settings["api_key"] and not self.settings['use_proxy']):
-        #     self.owm = OWM(self.settings["api_key"])
-        # else:
-        #     self.owm = OWMApi()
-        self.owm = OWMApi()
+        # lat,lon parameters are handled by OWMApi() both for
+        # local api key and mycrodt's api
+
+        self.owm = OWMApi( self.settings["api_key"], self.settings['use_proxy'] )
```

Comment:
What's the thinking behind two settings? Sometimes people with their own API key may wish to switch on/off the proxy without adding and removing the key?

Also need to add these settings to the `settingsmeta.json`

Reply:
You explain that to me, I just took the parameters that were handled before - I think the proxy should be handled somehow much more globally.

Reply:
Ah I didn't see that it was just un-commenting some code. This was before my time, not sure what the thinking was. 

Personally I think the less settings the better unless it's adding real value, so if there was no need for a separate setting I would leave it at just an `api_key`. If the custom key is present then use a local OWM connection, else use the proxy.